runtime/onert/core/include/compiler/StaticShapeInferer.h

```diff
@@ -28,6 +28,36 @@ namespace onert
 {
 namespace compiler
 {
+/**
+ * @brief Class that observe and update operands.
+ */
+class OperandObserver
```

Comment:
I thought it uses observer pattern from naming, but it seems operand set wrapper for update (operand info updater?), not observer.

Reply:
> I thought it uses observer pattern from naming, but it seems operand set wrapper for update (operand info updater?), not observer.

Right, it's not strictly observer pattern. But it performs a role similar to observer in observer pattern. `StaticShapeInferer` lets `OperandObserver`s register with a `StaticShapeInferer` and notifies registered `OperandObserver`s to update operand info. The difference from observer pattern is that it does not provide a observer interface and handles only `OperandObserver`.

Reply:
> I reviewed StaticShaperInferer.h and part of Compiler.cc today. It takes too long time to understand and review. I'll check remain part tomorrow.

OK. This is not urgent. Thanks for your review.

Reply:
>But it performs a role similar to observer in observer pattern. StaticShapeInferer lets OperandObservers register with a StaticShapeInferer and notifies registered OperandObservers to update operand info. The difference from observer pattern is that it does not provide a observer interface and handles only OperandObserver.

I understood. Thanks. It's OK.